common-utils/src/main/java/eu/arrowhead/common/mqtt/MqttServlet.java

```diff
@@ -0,0 +1,87 @@
+package eu.arrowhead.common.mqtt;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.LinkedBlockingQueue;
+
+import org.eclipse.paho.client.mqttv3.MqttMessage;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+import org.springframework.util.Assert;
+
+import eu.arrowhead.common.Constants;
+import eu.arrowhead.common.Utilities;
+import eu.arrowhead.common.mqtt.handler.MqttTopicHandler;
+
+@Component
+@ConditionalOnProperty(name = Constants.MQTT_API_ENABLED, matchIfMissing = false)
+public class MqttServlet {
```

Comment:
I'd rename this to MqqtDispatcher, because it is not a Servlet. Does not implement the Servlet interface.